spring-integration-gemfire/src/main/java/org/springframework/integration/gemfire/store/GemfireMessageStore.java

```diff
@@ -58,41 +50,37 @@
 	 * @param messageStoreRegion The region.
 	 */
 	public GemfireMessageStore(Region<Object, Object> messageStoreRegion) {
-		this.cache = null;
+		this(messageStoreRegion, "");
+	}
+
+	/**
+	 * Construct the {@link GemfireMessageStore} instance based on the provided
+	 * @param messageStoreRegion the region to use.
+	 * @param prefix the prefix to use.
+	 * @since 4.3.12
+	 */
+	public GemfireMessageStore(Region<Object, Object> messageStoreRegion, String prefix) {
+		super(prefix);
+		Assert.notNull(messageStoreRegion, "'messageStoreRegion' must not be null");
 		this.messageStoreRegion = messageStoreRegion;
 	}
 
+	/**
+	 * The boolean flag to ignore JTA on the Gemfire Region.
+	 * @param ignoreJta boolean flag to ignore JTA on the Gemfire Region.
+	 * @deprecated with no-op, in favor of externally configured region.
+	 */
+	@Deprecated
```

Comment:
There is no discussion about this change in the commit comment.

If we're deprecating in 4.3.12, shouldn't we remove it in 5.0?

Reply:
Yeah... Really my fault about that description. I've just violated my own rule about commits! 
Shame on me!


How about this?

```
* Deprecate the `GemfireMessageStore` `Cache`-based configuration - `setIgnoreJta()` and `afterPropertiesSet()`.
The `GemfireMessageStore` relies only on externally configured `Region`.
```

Yes, I absolutely agree to remove the stuff in `5.0`. 
One more my fault do not mention it at least in the comments for the PR.

Reply:
Thanks; I will add that and merge (and remove on master).